src/ripple/rpc/handlers/AccountLines.cpp

```diff
@@ -93,9 +93,12 @@ Json::Value doAccountLines (RPC::Context& context)
     std::string strIdent (params[jss::account].asString ());
     AccountID accountID;
 
-    // Intentional assignment in if.  Extra parentheses silence warning.
-    if ((result = RPC::accountFromString (accountID, strIdent)))
+    if (auto jv = RPC::accountFromString (accountID, strIdent))
+    {
+        for (auto it = jv.begin (); it != jv.end (); ++it)
```

Comment:
That's up to you. Although, I wouldn't bother. I would just wait for the Range TS in the stdlib. Its only a few years away.